master/custom/factories.py

```diff
@@ -1219,3 +1219,68 @@ def setup(self, **kwargs):
     def host_triple(props):
         build_triple = props.getProperty("build_triple")
         return build_triple.split("-")[0] + "-linux-android"
+
+
+class ValgrindBuild(UnixBuild):
+    buildersuffix = ".valgrind"
+    configureFlags = [
+        "--with-pydebug",
+        "--with-valgrind",
+        "--without-pymalloc",
+    ]
+    testFlags = [
+        "test_grammar",
+        "test_syntax",
+        "test_tokenize",
+        "test_fstring",
+        "test_ast",
+        "test_exceptions",
+    ]
+    factory_tags = ["valgrind"]
+    test_timeout = TEST_TIMEOUT * 5
+
+    def setup(self, parallel, branch, **kwargs):
+        self.addStep(
+            Configure(
+                command=["./configure", "--prefix", "$(PWD)/target"] + self.configureFlags
+            )
+        )
+        
+        compile = ["make", self.makeTarget]
+        if parallel:
+            compile = ["make", parallel, self.makeTarget]
+        
+        self.addStep(Compile(command=compile, env=self.compile_environ))
+        
+        self.addStep(
+            ShellCommand(
+                name="pythoninfo",
+                description="pythoninfo",
+                command=["make", "pythoninfo"],
+                warnOnFailure=True,
+                env=self.test_environ,
+            )
+        )
+        
+        test = [
+            "valgrind",
+            "--leak-check=full",
+            "--show-leak-kinds=definite",
+            "--error-exitcode=1",
```

Comment:
Can you use a larger value, such as 10? regrtest uses exit codes from 1 to 5.

Reply:
Yup!

Reply:
Done